lib/matplotlib/tests/test_legend.py

```diff
@@ -1667,3 +1667,120 @@ def test_boxplot_legend_labels():
+
+    plt.close(fig)
+
+
+def test_patchcollection_legend_properties():
+    # Test that PatchCollection legend preserves visual properties
+    fig, ax = plt.subplots()
+
+    # Create patches with specific properties
+    p1 = mpatches.Circle((0, 0), 1)
+    p2 = mpatches.Circle((2, 0), 1)
+
+    # Create a PatchCollection with specific visual properties
+    pc = mcollections.PatchCollection(
+        [p1, p2],
+        label="styled patches",
+        facecolor='red',
+        edgecolor='blue',
+        linewidths=3,
+        linestyle='--'
+    )
+    ax.add_collection(pc)
+    ax.autoscale_view()
+
+    # Create the legend
+    leg = ax.legend()
+    legend_patch = leg.legend_handles[0]
+
+    # Verify that visual properties are preserved
+    assert_allclose(legend_patch.get_facecolor()[:3],
+                    pc.get_facecolor()[0][:3], rtol=1e-5)
+    assert_allclose(legend_patch.get_edgecolor()[:3],
+                    pc.get_edgecolor()[0][:3], rtol=1e-5)
+    assert legend_patch.get_linewidth() == pc.get_linewidths()[0]
+    assert legend_patch.get_linestyle() == pc.get_linestyles()[0]
+
+    plt.close(fig)
+
+
+def test_patchcollection_legend_match_original():
+    # Test PatchCollection legend with match_original=True
+    fig, ax = plt.subplots()
+
+    # Create patches with individual colors
+    p1 = mpatches.Rectangle((0, 0), 1, 1, facecolor='red', edgecolor='black')
+    p2 = mpatches.Rectangle((2, 0), 1, 1, facecolor='green', edgecolor='blue')
+
+    # Create a PatchCollection with match_original=True
+    # This should use the first patch's colors
+    pc = mcollections.PatchCollection([p1, p2], match_original=True,
+                                      label="original colors")
+    ax.add_collection(pc)
+    ax.autoscale_view()
+
+    # Create the legend
+    leg = ax.legend()
+    legend_patch = leg.legend_handles[0]
+
+    # The legend should show the first patch's colors (red face, black edge)
+    # Note: colors are normalized to RGBA tuples
+    assert_allclose(legend_patch.get_facecolor()[:3],
+                    pc.get_facecolor()[0][:3], rtol=1e-5)
+    assert_allclose(legend_patch.get_edgecolor()[:3],
+                    pc.get_edgecolor()[0][:3], rtol=1e-5)
+
+    plt.close(fig)
+
+
+def test_patchcollection_legend_empty():
+    # Test that empty PatchCollection doesn't crash
+    fig, ax = plt.subplots()
+
+    # Create an empty PatchCollection
+    pc = mcollections.PatchCollection([], label="empty collection")
+    ax.add_collection(pc)
+    ax.set_xlim(0, 1)
+    ax.set_ylim(0, 1)
+
+    # This should not crash
+    leg = ax.legend()
+
+    # Check that the label still appears
+    assert len(leg.get_texts()) == 1
+    assert leg.get_texts()[0].get_text() == "empty collection"
+
+    # The legend handle should exist (with default/transparent colors)
+    assert len(leg.legend_handles) == 1
+    assert isinstance(leg.legend_handles[0], mpatches.Rectangle)
```

Comment:
Do we want to leave this out? I'd claim that being a `Rectangle` is an implementation detail that we don't want to test.

Instead we should check the color.

Reply:
I've removed the `isinstance(Rectangle)` checks from both tests.

In `test_patchcollection_legend`, I replaced it with color verification - now checking that the legend patch has the correct facecolor and edgecolor.
In `test_patchcollection_legend_empty`, I simply verify that the handle exists since the main goal is to test that empty collections don't crash.

This way the tests focus on actual behavior (correct colors) rather than implementation details (patch type).